AK/Vector.h

```diff
@@ -41,6 +42,8 @@ class VectorIterator {
     }
     VectorIterator operator-(int value) { return { m_vector, m_index - value }; }
     VectorIterator operator+(int value) { return { m_vector, m_index + value }; }
+    VectorIterator operator+(const VectorIterator& other) { return { m_vector, m_index + other.m_index }; }
+    VectorIterator operator/(int value) { return { m_vector, m_index / value }; }
```

Comment:
Overloading `operator/` for an iterator feels really bizarre. This is not an API that normal users of `VectorIterator` need, so let's do this with offset arithmetic instead (iterators can be subtracted from one another to give you an `int`, which can be added to `begin()`.)

Also, ideally we would implement this using only `operator<`, so that the algorithm is available for any class that provides a less-than operator